myhoard/controller.py

```diff
@@ -1086,10 +1074,9 @@ def _get_upload_backup_site(self):
             raise Exception("Defined upload site not present in list of non-recovery backup sites")
         return site_id, non_recovery_sites[site_id]
 
-    def _get_site_for_stream_id(self, stream_id):
-        for backup in self.state["backups"]:
-            if backup["stream_id"] == stream_id:
-                return backup["site"]
+    def _get_site_for_stream_id(self, stream_id: str) -> str:
+        if stream_id in self.state["backups"]:
+            return self.state["backups"][stream_id]["site"]
```

Comment:
I didn't check what threads this is accessed from but probably to avoid any future problems it would anyway be better to put this if within a `with self.lock:` block because access of `"backups"` is no longer atomic with this change.